packages/auth-sveltekit/src/server.ts

```diff
@@ -577,6 +685,124 @@ async function handleAuthRoutes(
       });
     }
 
+    case "magiclink/callback": {
+      if (!onMagicLinkCallback) {
+        throw new ConfigurationError(
+          `'onMagicLinkCallback' auth route handler not configured`
+        );
+      }
+
+      const error = searchParams.get("error");
+      if (error) {
+        const desc = searchParams.get("error_description");
+        return onMagicLinkCallback({
+          error: new MagicLinkFailureError(error + (desc ? `: ${desc}` : "")),
+        });
+      }
+
+      const code = searchParams.get("code");
+      const isSignUp = searchParams.get("isSignUp") === "true";
+      const verifier = cookies.get(config.pkceVerifierCookieName);
+      if (!code) {
+        return onMagicLinkCallback({
+          error: new PKCEError("no pkce code in response"),
+        });
+      }
+
+      if (!verifier) {
+        return onMagicLinkCallback({
+          error: new PKCEError("no pkce verifier cookie found"),
+        });
+      }
+
+      let tokenData: TokenData;
+      try {
+        tokenData = await (await core).getToken(code, verifier);
+      } catch (err) {
+        return onMagicLinkCallback({
+          error: err instanceof Error ? err : new Error(String(err)),
+        });
+      }
+
+      cookies.set(config.authCookieName, tokenData.auth_token, {
+        httpOnly: true,
+        sameSite: "lax",
+        path: "/",
+      });
+
+      cookies.set(config.pkceVerifierCookieName, "", {
+        maxAge: 0,
+        path: "/",
+      });
+
+      return onMagicLinkCallback({
+        error: null,
+        tokenData,
+        isSignUp,
+      });
+    }
+
+    case "webauthn/signup/options": {
+      const email = searchParams.get("email");
+      if (!email) {
+        throw new InvalidDataError("email missing");
+      }
+
+      return redirect(303, (await core).getWebAuthnSignupOptionsUrl(email));
```

Comment:
I think the other frameworks use 302 Found here?

Reply:
@scotttrinh  is 302 default if u don't provide anything, because svelte enforces you to provide the status... I was researching this while ago and I used 303 in all svelte examples, https://github.com/sveltejs/kit/issues/5976, https://www.drlinkcheck.com/blog/http-redirects-301-302-303-307-308
Now I found this links and I would still go with 303 hm?

Reply:
Hmm, in that case I think `307` is more accurate here. I know these are already GET requests, but I don't think we are intentionally trying to imply that you should always make a GET request here. I think the intention here is "You made some HTTP request to this address, but make the same request again to this new address."

Reply:
hmm makes sense